misk-crypto/src/main/kotlin/misk/crypto/ExternalKeyResolver.kt

```diff
@@ -0,0 +1,45 @@
+package misk.crypto
+
+import com.google.inject.Inject
+import wisp.logging.getLogger
+
+/**
+ * [ExternalKeyResolver] implements an [KeyResolver] that fetches Tink keysets from an external
+ * source, such as an S3 bucket. If multiple sources are registered (by binding implementations
+ * of [KeyResolver]), the first one to contain the key (via [ExternalKeySource.keyExists]) is
+ * the key that is used.
+ *
+ * If a key is not found, an [ExternalKeyManagerException] exception is raised.
+ */
+class ExternalKeyResolver @Inject constructor(
+  @ExternalDataKeys override val allKeyAliases: Map<KeyAlias, KeyType>,
+
+  private val externalKeySources: Set<ExternalKeySource>,
+) : KeyResolver {
+
+  private fun getRemoteKey(alias: KeyAlias): Key {
+    for (keySource in externalKeySources) {
+      logger.info("checking ${keySource.javaClass.name} for $alias")
+      if (keySource.keyExists(alias)) {
+        keySource.getKey(alias)?.let { return it }
+        logger.warn("$alias is reported to exist, but not able to retrieve; continuing")
```

Comment:
WDYT about changing the line above to `keySource.getKey(alias)!!`?
If the key isn't retrieved even though it exists, I think it's better to report that error instead of hiding it behind a more general warning log message.